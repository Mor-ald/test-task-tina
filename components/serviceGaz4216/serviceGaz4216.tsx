
import ServForm from "../servForm/ServForm";
import { Section } from "../util/section";

export const ServiceGaz4216 = (data) => {
    return (
      <Section color={"tint"}>
                    <div className="my-12 mx-24 flex flex-col">
                        {!data.serviceGaz4216.header || <p className='my-4 font-bold text-3xl'>{data.serviceGaz4216.header}</p>}
                        {!data.serviceGaz4216.time && !data.serviceGaz4216.coef && !data.serviceGaz4216.model || <p className="my-4">Цена услуги: <span className="font-bold text-orange-500">{((Number(data.serviceGaz4216.time.split(",").join(".")) * Number(data.serviceGaz4216.coef))*0.85).toFixed(0)} руб</span> (для Физ. лиц) Модель: <span className="font-bold text-orange-500">{data.serviceGaz4216.model}</span></p>}
                        <p className="my-4">
                            Специалисты сети автомастерских 
                            «АРМ» готовы выполнить любые работы 
                            по ремонту вашей ГАЗели и Лада, 
                            а также замене любых комплектующих 
                            и узлов по качеству не уступающему 
                            ремонту у дилера, при сохранении гораздо 
                            более привлекательной цены на работу.
                        </p>
                        {!data.serviceGaz4216.header && !data.serviceGaz4216.model || 
                            <p>
                                Мы приглашаем вас записаться на 
                                бесплатный осмотр вашей ГАЗели и 
                                Лада по <span className="font-bold text-orange-500">41 параметру</span> в одной из 
                                наших СТО, наиболее подходящей вам территориально. 
                                Для записи на <span className="font-bold text-orange-500">{data.serviceGaz4216.header}</span> 
                                <span className="font-bold text-orange-500"> для {data.serviceGaz4216.model} </span> 
                                звоните нам на удобную для вас <span className="font-bold text-orange-500">СТО</span>!
                            </p>
                        }
                        <ServForm visibleModel={false} serv={data.serviceGaz4216.header ? data.serviceGaz4216.header:""} model={"Газель 4216"}></ServForm>
                    </div>
      </Section>
    );
};