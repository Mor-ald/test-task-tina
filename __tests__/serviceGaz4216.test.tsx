/* eslint-disable react/react-in-jsx-scope */

import { ServiceGaz4216 } from "../components/serviceGaz4216/serviceGaz4216";

import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("ServiceGaz4216 component", () => {
	const data = {
		serviceGaz4216: {
			header: "Test Header",
			time: "1,5",
			coef: "1000",
			model: "Test Model",
		},
	};

	const component = render(<ServiceGaz4216 {...data} />); 
    
	it("renders without crashing", () => {
		render(<ServiceGaz4216 {...data} />);
		expect(component).toMatchSnapshot();
	});
  

});