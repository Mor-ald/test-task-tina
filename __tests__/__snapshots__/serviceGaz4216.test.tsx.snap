// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceGaz4216 component renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex-1 relative transition duration-150 ease-out body-font overflow-hidden text-gray-900 dark:text-gray-100 bg-gradient-to-br from-gray-100 dark:from-gray-1000 to-transparent "
      >
        <div
          class="my-12 mx-24 flex flex-col"
        >
          <p
            class="my-4 font-bold text-3xl"
            data-testid="header"
          >
            Test Header
          </p>
          <p
            class="my-4"
          >
            Цена услуги: 
            <span
              class="font-bold text-orange-500"
              data-testid="price"
            >
              1275
               руб
            </span>
             (для Физ. лиц) Модель: 
            <span
              class="font-bold text-orange-500"
              data-testid="model"
            >
              Test Model
            </span>
          </p>
          <p
            class="my-4"
            data-testid="spec"
          >
            Специалисты сети автомастерских «АРМ» готовы выполнить любые работы по ремонту вашей ГАЗели и Лада, а также замене любых комплектующих и узлов по качеству не уступающему ремонту у дилера, при сохранении гораздо более привлекательной цены на работу.
          </p>
          <p
            data-testid="welcome"
          >
            Мы приглашаем вас записаться на бесплатный осмотр вашей ГАЗели и Лада по 
            <span
              class="font-bold text-orange-500"
            >
              41 параметру
            </span>
             в одной из наших СТО, наиболее подходящей вам территориально. Для записи на 
            <span
              class="font-bold text-orange-500"
            >
              Test Header
            </span>
            <span
              class="font-bold text-orange-500"
            >
               для 
              Test Model
               
            </span>
            звоните нам на удобную для вас 
            <span
              class="font-bold text-orange-500"
            >
              СТО
            </span>
            !
          </p>
          <div>
            <div
              class="p-toast p-component p-toast-top-right p-ripple-disabled"
            >
              <div />
            </div>
            <form
              class="my-6 flex flex-col"
              data-testid="servForm"
            >
              <div
                class="my-2 flex flex-col"
              >
                <span
                  class="mb-2"
                >
                  <span
                    class="text-orange-500 font-bold"
                  >
                    *
                  </span>
                  Имя
                </span>
                <span
                  class="p-input-icon-left"
                >
                  <i
                    class="pi pi-user"
                  />
                  <input
                    class="p-inputtext p-component"
                    data-testid="nameUser"
                    name="nameUser"
                    placeholder="Имя"
                    required=""
                    value=""
                  />
                </span>
              </div>
              <div
                class="my-2 flex flex-col"
              >
                <span
                  class="mb-2"
                >
                  <span
                    class="text-orange-500 font-bold"
                  >
                    *
                  </span>
                  Телефон
                </span>
                <span
                  class="p-input-icon-left"
                >
                  <i
                    class="pi pi-phone"
                  />
                  <input
                    class="p-inputtext p-component p-inputmask"
                    data-testid="phone"
                    name="phone"
                    placeholder="Телефон"
                    required=""
                    type="phone"
                  />
                </span>
              </div>
              <div
                class="my-2 flex items-center"
              >
                <div
                  class="p-checkbox p-component"
                  data-testid="agreement"
                >
                  <div
                    class="p-hidden-accessible"
                  >
                    <input
                      id="agreement"
                      name="agreement"
                      required=""
                      type="checkbox"
                    />
                  </div>
                  <div
                    class="p-checkbox-box"
                  >
                    <span
                      class="p-checkbox-icon p-c"
                    />
                  </div>
                </div>
                <label
                  class="ml-2"
                  for="agreement"
                >
                  <span
                    class="text-orange-500 font-bold"
                  >
                    *
                  </span>
                  Я согласен на обработку моих 
                  <a
                    class="text-orange-500 font-bold underline"
                    href="/agreement"
                  >
                    персональных данных
                  </a>
                  , политикой конфеденциальности и договором оферты.
                </label>
              </div>
              <div>
                <button
                  aria-label="Отправить"
                  class="p-button p-component"
                  style="background: rgb(234, 108, 69); border: 0px; margin-top: 24px; padding: 12px 24px; font-weight: bold;"
                  type="submit"
                >
                  <span
                    class="p-button-icon p-c p-button-icon-left pi pi-check"
                  />
                  <span
                    class="p-button-label p-c"
                  >
                    Отправить
                  </span>
                </button>
              </div>
            </form>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex-1 relative transition duration-150 ease-out body-font overflow-hidden text-gray-900 dark:text-gray-100 bg-gradient-to-br from-gray-100 dark:from-gray-1000 to-transparent "
    >
      <div
        class="my-12 mx-24 flex flex-col"
      >
        <p
          class="my-4 font-bold text-3xl"
          data-testid="header"
        >
          Test Header
        </p>
        <p
          class="my-4"
        >
          Цена услуги: 
          <span
            class="font-bold text-orange-500"
            data-testid="price"
          >
            1275
             руб
          </span>
           (для Физ. лиц) Модель: 
          <span
            class="font-bold text-orange-500"
            data-testid="model"
          >
            Test Model
          </span>
        </p>
        <p
          class="my-4"
          data-testid="spec"
        >
          Специалисты сети автомастерских «АРМ» готовы выполнить любые работы по ремонту вашей ГАЗели и Лада, а также замене любых комплектующих и узлов по качеству не уступающему ремонту у дилера, при сохранении гораздо более привлекательной цены на работу.
        </p>
        <p
          data-testid="welcome"
        >
          Мы приглашаем вас записаться на бесплатный осмотр вашей ГАЗели и Лада по 
          <span
            class="font-bold text-orange-500"
          >
            41 параметру
          </span>
           в одной из наших СТО, наиболее подходящей вам территориально. Для записи на 
          <span
            class="font-bold text-orange-500"
          >
            Test Header
          </span>
          <span
            class="font-bold text-orange-500"
          >
             для 
            Test Model
             
          </span>
          звоните нам на удобную для вас 
          <span
            class="font-bold text-orange-500"
          >
            СТО
          </span>
          !
        </p>
        <div>
          <div
            class="p-toast p-component p-toast-top-right p-ripple-disabled"
          >
            <div />
          </div>
          <form
            class="my-6 flex flex-col"
            data-testid="servForm"
          >
            <div
              class="my-2 flex flex-col"
            >
              <span
                class="mb-2"
              >
                <span
                  class="text-orange-500 font-bold"
                >
                  *
                </span>
                Имя
              </span>
              <span
                class="p-input-icon-left"
              >
                <i
                  class="pi pi-user"
                />
                <input
                  class="p-inputtext p-component"
                  data-testid="nameUser"
                  name="nameUser"
                  placeholder="Имя"
                  required=""
                  value=""
                />
              </span>
            </div>
            <div
              class="my-2 flex flex-col"
            >
              <span
                class="mb-2"
              >
                <span
                  class="text-orange-500 font-bold"
                >
                  *
                </span>
                Телефон
              </span>
              <span
                class="p-input-icon-left"
              >
                <i
                  class="pi pi-phone"
                />
                <input
                  class="p-inputtext p-component p-inputmask"
                  data-testid="phone"
                  name="phone"
                  placeholder="Телефон"
                  required=""
                  type="phone"
                />
              </span>
            </div>
            <div
              class="my-2 flex items-center"
            >
              <div
                class="p-checkbox p-component"
                data-testid="agreement"
              >
                <div
                  class="p-hidden-accessible"
                >
                  <input
                    id="agreement"
                    name="agreement"
                    required=""
                    type="checkbox"
                  />
                </div>
                <div
                  class="p-checkbox-box"
                >
                  <span
                    class="p-checkbox-icon p-c"
                  />
                </div>
              </div>
              <label
                class="ml-2"
                for="agreement"
              >
                <span
                  class="text-orange-500 font-bold"
                >
                  *
                </span>
                Я согласен на обработку моих 
                <a
                  class="text-orange-500 font-bold underline"
                  href="/agreement"
                >
                  персональных данных
                </a>
                , политикой конфеденциальности и договором оферты.
              </label>
            </div>
            <div>
              <button
                aria-label="Отправить"
                class="p-button p-component"
                style="background: rgb(234, 108, 69); border: 0px; margin-top: 24px; padding: 12px 24px; font-weight: bold;"
                type="submit"
              >
                <span
                  class="p-button-icon p-c p-button-icon-left pi pi-check"
                />
                <span
                  class="p-button-label p-c"
                >
                  Отправить
                </span>
              </button>
            </div>
          </form>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
